src/EnergyPlus/GroundTemperatureModeling/FiniteDifferenceGroundTemperatureModel.cc

```diff
@@ -100,7 +100,7 @@ FiniteDiffGroundTempsModel::FiniteDiffGTMFactory(EnergyPlusData &state, GroundTe
 
     // Search through finite diff models here
     std::string const cCurrentModuleObject =
-        state.dataGrndTempModelMgr->CurrentModuleObjects(static_cast<int>(GroundTempObjType::FiniteDiffGroundTemp));
+        static_cast<std::string>(groundTempModelNamesUC[static_cast<int>(GroundTempObjType::FiniteDiffGroundTemp)]);
```

Comment:
All of these ground temp models are hanging out on the `EnergyPlus` namespace. I should probably fix that at some point. Also, all of this business with pointers should probably be cleaned up at some point, too. 2015 @mitchute was too clever by half, it seems.

Reply:
Don't cast this to a `std::string`. Keep `cCurrentModuleObject` as a `std::string_view` (input processor routines should accept it). You're not the only the one on that corner, there are all sorts of things in EnergyPlus that I would categorize as "too clever by half".